Hpcc 15612 Embedded Couchbase Plugin

rpastrana opened the pull request.

rpastrana commented:
@jamienoss please review

rpastrana commented:
Not squashing the multiple commits for history keeping. To review, use "Files changed" tab.

HPCCSmoketest commented:
Automated Smoketest 
Sha: eca8fb5cf2c1d64063eb8870683afd8d7afcdba1
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: 19ef258eb55c7ae5d056f21c08a92a2d202f5de3
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: 1e5413590156e1468f80400902c3181cc109090d
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

rpastrana committed:
HPCC-15612 Libcouchbase as an embedded language in ECL

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest 
Sha: b36e8d94ec7fe0b4a6df5df026373c6d8843337f
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented:
@xwang2713 we will need to install libcouchbase on our build machines.

rpastrana commented:
@dcamper please review.

dcamper commented on cmake_modules/FindCOUCHBASE.cmake:
```diff
@@ -0,0 +1,93 @@
+      NO_DEFAULT_PATH
+    )
+
+    FIND_LIBRARY(
```
Is it correct to be finding MySQL libraries here?

dcamper commented on plugins/couchbase/couchbaseembed.hpp:
```diff
@@ -0,0 +1,390 @@
+        return count;
+    }
+
+    void handleDeserializeOutcome(DeserializationResult resultcode, const char * targetype, const char * culpritvalue)
```
Can this be static?

rpastrana replied:
Yes, and I will make it such...

dcamper commented on plugins/couchbase/couchbaseembed.hpp:
```diff
@@ -0,0 +1,390 @@
+        CouchbaseConnection(const CouchbaseConnection &);
+    };
+
+    class CouchbaseRowBuilder : public CInterfaceOf<IFieldSource>
```
The code for this class feels like it should be included in the .cpp file instead of the header.  Was there a dependency issue?

dcamper commented:
@rpastrana  A couple of minor notes and questions.  Overall it looks great.

rpastrana committed:
HPCC-15612 Code Review changes

- Searches for libcouchbase in externals folder
- Splits out class declarations and implementations
- Declares several static methods

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@dcamper please review changes based on your comments:  		29397b9

HPCCSmoketest commented:
Automated Smoketest 
Sha: 29397b9c6e93f058f1ee0c6c4d923b1f03c9660a
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:82 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented:
@Michael-Gardner please review build logic

Michael-Gardner committed:
Change to PR - Michael Gardner

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

rpastrana committed:
HPCC-15612 Merging M. Gardner's CMAKE changes

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@richardkchapman please review.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 1d93bcaf15ac720ebdf80aaad137bec8e217fc9e
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:82 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented on .gitmodules:
```diff
@@ -37,3 +37,6 @@
     url = https://github.com/hpcc-systems/lz4.git
+[submodule "plugins/couchbase/libcouchbase-cxx"]
+	path = plugins/couchbase/libcouchbase-cxx
+	url = https://github.com/couchbaselabs/libcouchbase-cxx.git
```
We should fork the couchbaselabs repo into hpcc-systems, and use our fork. Otherwise we would break should they decide to rename/move/delete/rebase their repo.

richardkchapman replied:
(I have done the fork - can you update your PR)

richardkchapman commented:
@rpastrana Looks good - one comment

rpastrana committed:
HPCC-15612 Target hpcc-systems' fork of libcouchbase-cxx

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

jamienoss commented:
@richardkchapman & @rpastrana am I still wanted to look over this?

HPCCSmoketest commented:
Automated Smoketest 
Sha: edc2fae1577335fc3a1e15d92e85e336813a1575
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:82 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented on system/jlib/jptree.cpp:
```diff
@@ -1057,7 +1057,7 @@ void PTree::setName(const char *_name)
         name = NULL;
     else
     {
-        if (!validateXMLTag(_name)) 
```
This unrelated change snuck in, and can't revert since the extra space doesn't pass our check-in criteria

richardkchapman replied:
No big deal, it can stay

jamienoss replied:
@rpastrana You can use "git commit -ns" to bypass this.

rpastrana commented:
@jamienoss, another set of eyes looking over this wouldn't hurt, but we are looking forward to closing this out soon. Let me know if you can take a look soon, otherwise I'll ask Richard to merge. Thanks.

jamienoss commented:
@rpastrana Sure thing, I'll take a look over it today.

jamienoss commented:
@rpastrana, @richardkchapman all looks good to me
